src/library/dlgtrackinfomulti.cpp

```diff
@@ -0,0 +1,833 @@
+#include "library/dlgtrackinfomulti.h"
+
+#include <QLineEdit>
+#include <QStyleFactory>
+#include <QtDebug>
+
+#include "defs_urls.h"
+#include "library/coverartcache.h"
+#include "library/coverartutils.h"
+#include "library/library_prefs.h"
+#include "moc_dlgtrackinfomulti.cpp"
+#include "preferences/colorpalettesettings.h"
+#include "sources/soundsourceproxy.h"
+#include "track/beatutils.h"
+#include "track/track.h"
+#include "util/color/color.h"
+#include "util/datetime.h"
+#include "util/duration.h"
+#include "util/optional.h"
+#include "widget/wcoverartlabel.h"
+#include "widget/wcoverartmenu.h"
+#include "widget/wstarrating.h"
+
+namespace {
+
+const QString kVariousText = char('<') + QObject::tr("various") + char('>');
+const char* kOrigValProp = "origVal";
+const QString kClearItem = QStringLiteral("clearItem");
+
+/// If value differs from the current value, add it to the list.
+/// If new and current are identical, keep only one. Later on we can use the
+/// item count to maybe join the list and format the font accordingly.
+/// List can be a QStringList or a QList<double>
+template<typename T>
+void appendToTagList(QList<T>* pList, const T& value) {
```

Comment:
it is not clear from the function name that this does special management to keep the list unique. Maybe "appendUniqueToTagList" or "addToTagSet" or "includeInTagSet" would make that more clear

Reply:
I agree with daniel's comment that these should be QSets, and then use an adapter to convert them into Lists when needed. it feels brittle otherwise

Reply:
Jup, that's exactly what I did. Had to deal with some type difficulties but managed it with `convertToQStringConvertible` from #13236

Reply:
"addToTagSet" sounds good.